client/components/date-picker/index.jsx

```diff
@@ -13,131 +13,131 @@ import DayItem from 'components/date-picker/day';
 
 /* Internal dependencies
  */
-module.exports = React.createClass( {
-	displayName: 'DatePicker',
-
-	propTypes: {
-		calendarViewDate: React.PropTypes.object,
-		enableOutsideDays: React.PropTypes.bool,
-		events: React.PropTypes.array,
-		locale: React.PropTypes.object,
-
-		selectedDay: React.PropTypes.object,
-		timeReference: React.PropTypes.object,
-
-		onMonthChange: React.PropTypes.func,
-		onSelectDay: React.PropTypes.func
-	},
-
-	getDefaultProps: function() {
-		return {
-			enableOutsideDays: true,
-			calendarViewDate: new Date(),
-			selectedDay: null,
-			onMonthChange: noop,
-			onSelectDay: noop
-		};
-	},
-
-	isSameDay: function( d0, d1 ) {
-		d0 = this.moment( d0 );
-		d1 = this.moment( d1 );
+class DatePicker extends PureComponent {
+	static propTypes = {
+		calendarViewDate: PropTypes.object,
+		enableOutsideDays: PropTypes.bool,
+		events: PropTypes.array,
+		locale: PropTypes.object,
+
+		selectedDay: PropTypes.object,
+		timeReference: PropTypes.object,
+
+		onMonthChange: PropTypes.func,
+		onSelectDay: PropTypes.func,
```

Comment:
`isSameDay` and `locale` also requires `this.props.moment` so it should be included in the propTypes.